src/components/Fieldset/Fieldset.tsx

```diff
@@ -0,0 +1,42 @@
+import React from 'react';
+import clsx from 'clsx';
+import './Fieldset.scss';
+import type {FieldsetProps} from './Fieldset.types';
+import {Typography} from '~/components';
+
+export const Fieldset = React.forwardRef<HTMLDivElement, FieldsetProps>(({
+    id,
+    label,
+    helper,
+    children,
+    className,
+    buttons,
+    ...props
+}, ref) => {
+    return (
+        <div
+            ref={ref}
+            id={id}
+            className={clsx(
+                'moonstone-fieldset',
+                'flexCol_nowrap',
+                className
+            )}
+            {...props}
+        >
+            <div className={clsx('flexRow_nowrap', 'flexFluid', 'alignCenter')}>
+                <div className="flexRow_nowrap flexFluid alignCenter">
```

Comment:
I think you can remove this `div`, you can add classes on `Typography` is it's needed